tests/src/test/java/org/eclipse/hono/tests/http/EventHttpIT.java

```diff
@@ -98,7 +98,7 @@ public void testEventIsRejectedForUnsupportedQosLevel(final VertxTestContext ctx
                 Buffer.buffer("hello"),
                 requestHeaders,
                 ResponsePredicate.status(HttpURLConnection.HTTP_BAD_REQUEST))
-                .onComplete(ctx.completing());
+                .onComplete(ctx.succeedingThenComplete());
     }
 
```

Comment:
Regarding `EventHttpIT` in general:
I've run the integration tests like this: `mvn verify -Prun-tests,jaeger -Ddocker.keepRunning -Dit.test=EventHttpIT -Dmsg.count=1`.

When looking at the HTTP adapter traces with tags `cache_hit=true credentials_type=x509-cert` (corresponding Jaeger UI URL: http://localhost:18080/search?limit=200&lookback=1h&service=hono-adapter-http&tags=%7B%22cache_hit%22%3A%22true%22%2C%22credentials_type%22%3A%22x509-cert%22%7D) there were 117 traces using current master but only 59 traces when using the code from this PR. The overall number of traces with `credentials_type=x509-cert` was the same (121).

This makes me wonder whether changes in this PR somehow lead to less cache hits.

Reply:
Hm, it is probably not a coincidence that this is close to half of what we had before, right?

Reply:
Not sure here yet. The issue can be further tracked down by just concentrating on the `EventHttpIT#testConnectSucceedsWithAutoProvisioning` test.

Reply:
Not sure if I understand what you mean ...

Reply:
It seems that when trying to get the same credentials mulitple times, in `CredentialsApiAuthProvider#getCredentialsForDevice` the given `SubjectDnCredentials` have the same `clientContext` JsonObject content (e.g. `{"client-certificate":"MIIC4TCC..."}`), but the HashCode of these JsonObjects seems to be different now with every `getCredentialsForDevice` invocation.
With the older vert.x version it was the same.

Reply:
It looks like this is caused by changes to the `JsonObject.copy()` behaviour.

For a tenant with auto-provisioning enabled, `TenantServiceBasedX509Authentication#getCredentials` will return a JsonObject with an `client-certificate` entry containing a byte array.
This JsonObject is then copied in `X509AuthProvider#getCredentials`, and the copy is put into the `SubjectDnCredentials` object. The hashcode of that copy is then relevant for cache hit or miss.

Consider this example:
```
            final JsonObject authInfo = new JsonObject().put("client-certificate", new byte[] { 0x00, 0x01 })
                    .copy();
            System.out.println("hashcode: " + authInfo.hashCode() + "; map: " + authInfo.getMap());

            final JsonObject authInfo2 = new JsonObject().put("client-certificate", new byte[] { 0x00, 0x01 })
                    .copy();
            System.out.println("hashcode: " + authInfo2.hashCode() + "; map: " + authInfo2.getMap());
```

Output with vert.x 3.9.7:
```
hashcode: -207395283; map: {client-certificate=AAE=}
hashcode: -207395283; map: {client-certificate=AAE=}
```
With vert.x 4.1.2:
```
hashcode: -1623841546; map: {client-certificate=[B@6c8bca63}
hashcode: -864800432; map: {client-certificate=[B@3fc9dfc5}
```

Reply:
Yes, it seems that when copying a binary value, 3.9.x performed a Base64 encoding of the byte array whereas 4.1.x simply copies the byte array itself. I have added a *nomalization* step to the Credentials client where the caching key is created,